pkg/compactor/bucket_compactor.go

```diff
@@ -670,6 +676,10 @@ func NewBucketCompactorMetrics(blocksMarkedForDeletion prometheus.Counter, reg p
 			Name: "cortex_compactor_group_compactions_total",
 			Help: "Total number of group compaction attempts that resulted in new block(s).",
 		}),
+		blockCompactionDelay: promauto.With(reg).NewSummaryVec(prometheus.SummaryOpts{
```

Comment:
What do you think about using native histogram for this? It would give us high-precision data. With summary, we only get average (over time window).

Reply:
Average is all I had in mind, but a histogram would be fine as well. Do you have any buckets in mind?

Reply:
I would use native histogram. See https://grafana.com/docs/mimir/latest/send/native-histograms/#instrument-application-with-prometheus-client-libraries for an example and explanation of parameters.

Reply:
Sure, let me know what you think of the buckets I chose - they range from 1 minute to 2 hours.

Reply:
I think 1 minute is way too optimistic :) I would start at 10 minutes, and go up to several hours.

However since we have native histogram, we don't need to worry about buckets too much. Native histogram will only use buckets with observations and create any buckets it needs. Using bucket factor 1.1 looks good here. (Bucket boundaries will be powers of `1.090507733`).

Reply:
I changed the metric to minutes since seconds is too granular. Some better buckets, representing minutes: `[]float64{1.0, 5.0, 10.0, 30.0, 60.0, 120.0, 180.0, 240.0, 300.0, 600.0, 1200.0}`